### ISSUE TITLE
.fai not found due to symlink resolving

### ISSUE THREAD

**Issue by DavyCats** [Illumina/strelka#61]
It appears that strelka resolves symlinks pointing to a reference fasta before determining what the path to the fasta index is.  This is given us some trouble because our fasta file is under git-annex and doesn't have an index at the location of the actual file itself.

The error message which pointed this out to us:
```
[2018-08-09T12:05:50.824958Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] Error Message:
[2018-08-09T12:05:50.824985Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] Anomalous task wrapper stderr output. Wrapper signal file: '/exports/sasc/testing/workspace/Biowdl/somatic-variantcalling/BIOWDL-72/8/biowdl.test.strelka.StrelkaTestPaired/scatters/scatter-0.bed_runStrelka/workspace/pyflow.data/logs/tmp/taskWrapperLogs/000/008/pyflowTaskWrapper.signal.txt'
[2018-08-09T12:05:50.825010Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] Logging 7 line(s) of task wrapper log output below:
[2018-08-09T12:05:50.825034Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [taskWrapper-stderr] [2018-08-09T12:05:49.488334Z] [swellshark.researchlumc.nl] [10197_1] [pyflowTaskWrapper:CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [wrapperSignal] wrapperStart
[2018-08-09T12:05:50.825057Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [taskWrapper-stderr] [2018-08-09T12:05:49.513736Z] [swellshark.researchlumc.nl] [10197_1] [pyflowTaskWrapper:CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [wrapperSignal] taskStart
[2018-08-09T12:05:50.825078Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [taskWrapper-stderr] [2018-08-09T12:05:49.926604Z] [swellshark.researchlumc.nl] [10197_1] [pyflowTaskWrapper:CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [wrapperSignal] taskExitCode -11
[2018-08-09T12:05:50.825104Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [taskWrapper-stderr] [2018-08-09T12:05:49.931504Z] [swellshark.researchlumc.nl] [10197_1] [pyflowTaskWrapper:CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [wrapperSignal] taskStderrTail 3
[2018-08-09T12:05:50.825127Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [taskWrapper-stderr] Last 2 stderr lines from task (of 2 total lines):
[2018-08-09T12:05:50.825148Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [taskWrapper-stderr] [2018-08-09T12:05:49.570837Z] [swellshark.researchlumc.nl] [10197_1] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] WARNING: theta parameter file was not given. Using internal theta values.
[2018-08-09T12:05:50.825172Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [taskWrapper-stderr] [2018-08-09T12:05:49.672922Z] [swellshark.researchlumc.nl] [10197_1] [CallGenome+callGenomeSegment_chromId_000_chr1_0000_to_chromId_001_chrM_0000] [E::fai_build3] Failed to open FASTA index /exports/sasc/testing/fixtures/.git/annex/objects/wq/50/SHA256E-s32800--33054965ee2c4e2a1fdf75a9d669a4c3f85d010841792b5a493d89f868693281/SHA256E-s32800--33054965ee2c4e2a1fdf75a9d669a4c3f85d010841792b5a493d89f868693281.fai : Permission denied
[2018-08-09T12:05:50.825194Z] [swellshark.researchlumc.nl] [10197_1] [WorkflowRunner] [ERROR] Failed to complete sub-workflow task: 'CallGenome' launched from master workflow, failed sub-workflow classname: 'CallWorkflow'
```

The configuration command used (as noted in the comment at the top of the run script):
```
'<shortened_path>/conda_env/bin/configureStrelkaSomaticWorkflow.py --normalBam <omitted> --tumorBam <omitted> --ref /exports/sasc/testing/workspace/Biowdl/somatic-variantcalling/BIOWDL-72/8/cromwell-executions/Strelka/baad5fa2-811d-4e94-9172-b5f834a77116/call-strelkaSomatic/shard-0/inputs/81721111/reference.fasta --runDir /exports/sasc/testing/workspace/Biowdl/somatic-variantcalling/BIOWDL-72/8/biowdl.test.strelka.StrelkaTestPaired/scatters//scatter-0.bed_runStrelka --callRegions <omitted>
```
Notice that the path to the file it fails does not match the file path given in the command.

**Comment by ctsa** [comment]
Thanks for providing a detailed description of the issue. I can reproduce this. We took care of this issue for the input bam paths a while ago, but it looks like a separate function was path-cleaning the reference fasta and still trying to canonicalize it. We'll provide a fixed build within a few days.

**Comment by ctsa** [comment]
Added fix in v2.9.7: https://github.com/Illumina/strelka/releases/tag/v2.9.7
